File: app/schemas.py
from pydantic import BaseModel
from typing import Optional

class MessageSchema(BaseModel):
    id: str
    content: str
    chat_id: str
    created_at: str
    is_system: Optional[bool]
    is_markdown: Optional[bool]

class ChatSchema(BaseModel):
    id: str
    title: Optional[str]
    status: str
    user_id: str
    created_at: str
    updated_at: str

class ChatWithLatestMessageSchema(BaseModel):
    chat: ChatSchema
    latest_message: Optional[MessageSchema]

File: nextjs/schemas.ts
// cr-genesys/nextjs/schemas.ts

export type Message = {
  id: string;
  content: string;
  chatId: string;
  isSystem: boolean;
  isMarkdown: boolean;
  sentToGenesys: boolean;
  genesysMessageId: string | null;
  messageType: string;
  createdAt: Date;
  updatedAt: Date;
};

export type Chat = {
  id: string;
  title: string | null;
  status: 'OPEN' | 'CLOSED';
  userId: string;
  createdAt: Date;
  updatedAt: Date;
  genesysOpenMessageSessionId: string | null;
  genesysOpenMessageActive: boolean;
};

export type ChatWithLatestMessage = {
  chat: Chat;
  latestMessage: Message | null;
};